hail/src/main/scala/is/hail/expr/types/physical/PLocus.scala

```diff
@@ -9,93 +9,23 @@ import is.hail.utils._
 import is.hail.variant._
 
 object PLocus {
-  def apply(rg: ReferenceGenome): PLocus = PLocus(rg.broadcastRG)
+  def apply(rg: ReferenceGenome): PLocus = PCanonicalLocus(rg.broadcastRG)
 
-  def apply(rg: ReferenceGenome, required: Boolean): PLocus = PLocus(rg.broadcastRG, required)
+  def apply(rg: ReferenceGenome, required: Boolean): PLocus = PCanonicalLocus(rg.broadcastRG, required)
 
-  def representation(required: Boolean = false): PStruct = PStruct(
-      required,
-      "contig" -> PString(required = true),
-      "position" -> PInt32(required = true))
-
-  def schemaFromRG(rg: Option[ReferenceGenome], required: Boolean = false): PType = rg match {
-    case Some(ref) => PLocus(ref)
-    case None => PLocus.representation(required)
-  }
+  def apply(rgBc: BroadcastRG, required: Boolean = false): PLocus = PCanonicalLocus(rgBc, required)
 }
 
-case class PLocus(rgBc: BroadcastRG, override val required: Boolean = false) extends ComplexPType {
-  def rg: ReferenceGenome = rgBc.value
+abstract class PLocus extends ComplexPType {
+  def rgBc: BroadcastRG
```

Comment:
left this only so that the virtual type signature could live on the abstract class

Reply:
yeah, this is right.

Reply:
wait, this is actually totally unnecessary. Anyone who wants to use should grab it from the virtual type. We can refactor this a bit to clean it up after this PR goes in.

Reply:
So then `lazy val virtualType: TLocus = TLocus(rgBc, required)` doesn't take rgBc?

Reply:
oh, hmm. leave for now, we can make this simpler later.